Tasks/Gradle/gradlesonar.ts

```diff
@@ -0,0 +1,46 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/sonarqube-common.d.ts" />
+
+import path = require('path');
+import fs = require('fs');
+
+import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
+
+// Lowercased file names are to lessen the likelihood of xplat issues
+import sqCommon = require('sonarqube-common/sonarqube-common');
+import {SonarQubeEndpoint} from 'sonarqube-common/sonarqube-common';
+
+// Apply arguments to enable SonarQube analysis
+export function applyEnabledSonarQubeArguments(gradleRun: trm.ToolRunner):trm.ToolRunner {
+    if (!tl.getBoolInput('sqAnalysisEnabled')) {
+        console.log("SonarQube analysis is not enabled");
+        return gradleRun;
+    }
+
+    console.log("SonarQube analysis is enabled");
```

Comment:
Yes, even log messages :). See other task, for example the CopyFiles task. Debug messages, i.e. tl.Debug() - do not seem to be localizable. Up to you to choose the verbosity of your log messages, but bear in mind that if a bug occurs, it's useful to be able to tell the user to enable system.debug and to give you his logs so that you can investigate further.